routes/projects.js

```diff
@@ -0,0 +1,85 @@
+const express = require('express');
+const db = require('../data/helpers/projectModel');
+
+
+function validateBody(arr) {
+    return function(req, res, next) {
+        try {
+            arr.map( (i)=>{
+                console.log(i);
+                if(!(typeof req.body[i]) || req.body[i] === "")
+                    throw "invalid request";
+            });
+            next();
+        }
+        catch(e){
+            res.status(400).json({errorMessage: e});
+        }
+    }
+}
+
+const router = express.Router();
+router.get('/', async (req,res) => {
+    try {
+       let result = await db.get();
+        if(result.length === 0) throw "no projects availible";
+        res.status(200).json(result);
+    }
+    catch(e) {
+        await res.status(500).json({errorMessage: e });
+    }
+});
+
+router.get('/:id', async (req,res) => {
+    try {
+        result = await db.get(req.params.id);
+        if((result || result.length === 0)) throw "project not availible";
```

Comment:
This check here is ambiguous, it should be if(!result || !result.length) or if(!result || result.length === 0). The way you have it set up it will always throw "project not available".

Reply:
yes that was a last minute change, I should have tested more.